tests/spicy/types/regexp/parse-captures-requires.spicy

```diff
@@ -0,0 +1,15 @@
+# @TEST-EXEC: printf 'babc' | spicy-driver -d %INPUT >output
+# @TEST-EXEC: btest-diff output
+#
+# @TEST-DOC: Check that we can use capture group from inside attributes
+
+module Test;
+
+public type X = unit {
+    a: /b/;
+    b: /a(b+)c/ &requires=($1 == self.a)
+          {
+              assert($1 == self.a);
```

Comment:
Nit: this came from the bug report, but we could simplify this slightly, e.g.,

```suggestion
    b: /a(b)c/ &requires=($1 == b"b")
          {
              assert($1 == b"b");
```

Reply:
Keeping the original version doesn't seem to hurt, might catch something else some time.